mimic/model/identity_objects.py

```diff
@@ -0,0 +1,305 @@
+"""
+Model objects for the Identity mimic.
+"""
+
+from __future__ import absolute_import, division, unicode_literals
+
+from zope.interface import implementer
+
+from twisted.plugin import IPlugin
+from twisted.web.http import BAD_REQUEST, FORBIDDEN, NOT_FOUND
+
+from mimic.catalog import Entry
+from mimic.catalog import Endpoint
+from mimic.imimic import IExternalAPIMock, IEndpointTemplate
+
+
+def _identity_error_message(msg_type, message, status_code, request):
+    """
+    Set the response code on the request, and return a JSON blob representing
+    a Identity error body, in the format Identity returns error messages.
+
+    :param str msg_type: What type of error this is - something like
+        "badRequest" or "itemNotFound" for Identity.
+    :param str message: The message to include in the body.
+    :param int status_code: The status code to set
+    :param request: the request to set the status code on
+
+    :return: dictionary representing the error body
+    """
+    request.setResponseCode(status_code)
+    return {
+        msg_type: {
+            "message": message,
+            "code": status_code
+        }
+    }
+
+
+def bad_request(message, request):
+    """
+    Return a 400 error body associated with a Identity bad request error.
+    Also sets the response code on the request.
+
+    :param str message: The message to include in the bad request body.
+    :param request: The request on which to set the response code.
+
+    :return: dictionary representing the error body.
+    """
+    return _identity_error_message("badRequest", message, BAD_REQUEST, request)
+
+
+def not_found(message, request):
+    """
+    Return a 404 error body associated with a Identity not found error.
+    Also sets the response code on the request.
+
+    :param str message: The message to include in the bad request body.
+    :param request: The request on which to set the response code.
+
+    :return: dictionary representing the error body.
+    """
+    return _identity_error_message("itemNotFound", message, NOT_FOUND, request)
+
+
+def forbidden(message, request):
+    """
+    Return a 403 error body associated with a Identity forbidden error.
+    Also sets the response code on the request.
+
+    :param str message: The message to include in the bad request body.
+    :param request: The request on which to set the response code.
+
+    :return: dictionary representing the error body.
+    """
+    return _identity_error_message("forbidden", message, FORBIDDEN, request)
+
+
+@implementer(IExternalAPIMock, IPlugin)
+class ExternalApiStore(object):
+    """
+    A :obj:`ExternalApiStore` is an object which provides the functionality to list
+    an API that is external to Mimic in the service catalog provided by Mimic.
+    """
```

Comment:
The `ExternalApiStore` represents a service (e.g Cloud Files)  and hosts all the endpoint templates for that service (e.g ORD vs IAD vs SYD...). By default the templates are disabled and *can* be enabled on a per-tenant basis, or they can be enabled *globally* for all tenants.

I'll update the description.